browser-interface/packages/shared/session/index.ts

```diff
@@ -32,18 +35,35 @@ export const deleteSession = async (userId?: string) => {
 export const retrieveLastSessionByAddress: (address: string) => Promise<StoredSession | null> = async (
   address: string
 ) => {
-  const sessions = (await retrieveAllCurrentSessions()).filter(withAddress(address))
-
-  return sessions.length > 0 ? sessions[0] : null
+  return getSessionWithSSOIdentity((await retrieveAllCurrentSessions()).filter(withAddress(address)))
 }
 
 /**
  * Retrieve the last session stored that is a guest session
  */
 export const retrieveLastGuestSession: () => Promise<StoredSession | null> = async () => {
-  const sessions = (await retrieveAllCurrentSessions()).filter(isGuest).sort(byExpiration)
+  return getSessionWithSSOIdentity((await retrieveAllCurrentSessions()).filter(isGuest).sort(byExpiration))
+}
+
+/**
+ * From a list of sessions return the first one with the SSO identity
+ */
+async function getSessionWithSSOIdentity(sessions: StoredSession[]) {
+  if (!sessions.length) {
+    return null
+  }
+
+  const session = sessions[0]
+
+  const ssoIdentity = await SingleSignOn.getIdentity(session.identity.address)
```

Comment:
Attempts to obtain the session from localstorage and the iframe. 
Source of truth is the iframe so if it doesn't have an identity it return null